src/app/components/appointmentform/actions.ts

```diff
@@ -0,0 +1,93 @@
+"use server";
+
+import {
+	Appointment,
+	createAppointment,
+	getAppointmentsBySchedule,
+	getAppointmentByID,
+} from "@/db/homeseeker/appointment";
+import { getScheduleByID } from "@/db/homeseeker/schedule";
+import { getCurrentUser } from "@/app/cookies";
+
+export async function makeAppointment(
+	schedule_id: number,
+	start_time: Date,
+	end_time: Date,
+): Promise<Appointment | string> {
+	// Errors checking
+	if (!start_time) {
+		return "Start time required.";
+	}
+	if (!end_time) {
+		return "End time required.";
+	}
```

Comment:
Don't know much much of a tall order this is (if it's too much I can just do it myself) but can we maybe use the `FormStatus` type and provide these as `fieldErrors`? There's an example of this in the `sign-in/actions.ts` file.

Main reason for this is that it allows us to show this validation data on the `TextInput` components.

Reply:
Honestly we dont need to check for that since in the page the time are set to required. The reason we didn't use the textinput along with the formstatus was something regarding: Type '"time"' is not assignable to type '"text" | "password" | "email" | "url" | undefined'.